manifest/morph/morph.go

```diff
@@ -626,7 +626,7 @@ func morphObjectToType(v tftypes.Value, t tftypes.Type, p *tftypes.AttributePath
 		// tftypes.NewValue() fails if any of the attributes in the object don't have a corresponding value
 		for k := range t.(tftypes.Object).AttributeTypes {
 			if _, ok := ovals[k]; !ok {
-				ovals[k] = tftypes.NewValue(t.(tftypes.Object).AttributeTypes[k], nil)
+				ovals[k], _ = newValue(t.(tftypes.Object).AttributeTypes[k], nil, p)
```

Comment:
I wouldn't suppress the diag here with `_`. I would check if there's a diag and return it.